src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+import java.util.Objects;
+
+public class MyHashMap<K,V> implements MyMap<K,V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K,V> bucket = table[Math.abs(hash(key)) % table.length];
+        Node<K,V> newNode = new Node<>(key, value, null);
+        if (bucket != null) {
+            while (bucket != null) {
+                if (Objects.equals(bucket.key, key)) {
+                    bucket.value = value;
+                    return;
+                }
+                bucket = bucket.next;
+            }
+            bucket = table[Math.abs(hash(key)) % table.length];
+            while (bucket.next != null) {
+                bucket = bucket.next;
+            }
+            bucket.next = newNode;
+            if (++size > threshold) {
+                resize();
```

Comment:
not fixed. the `resize` method must be called **at the beginning** of the method.

Reply:
I don't see anything wrong with calling resize at the end of a method, especially since java itself does it

Reply:
I'd say that putting a new node inside the array after the resize would be better at least not to put the same value 2 times and to have less collision. Imagine that we have like 1000 elements, or we have like 74 % of capacity filled, collision is at max and you are adding this last element, the amount of operations we could have avoided is quite big on top of that resize just doubles the work despite we already know it will be resized.
Regarding the original `HashMap`, I don't know the reasoning behind that decision, but we can also think about our reasoning and try to come up with some pros and cons